src/swap/SwapScreen.tsx

```diff
@@ -203,6 +210,36 @@ export function SwapScreen({ route }: Props) {
   const handleReview = () => {
     ValoraAnalytics.track(SwapEvents.swap_screen_review_swap)
 
+    if (useViemForSwap) {
+      if (!exchangeRate) {
+        // Error already shown, do nothing
+        return
+      }
+
+      if (parsedSwapAmount[Field.FROM].gt(fromTokenBalance)) {
+        setFromSwapAmountError(true)
+        showMaxCeloSwapWarning()
+        dispatch(showError(t('swapScreen.insufficientFunds', { token: fromToken?.symbol })))
+      }
+
+      const quoteResultType = exchangeRate.type
+      switch (quoteResultType) {
+        case 'need-decrease-swap-amount-for-gas': // fallthrough on purpose
+        case 'not-enough-balance-for-gas':
+          quoteResultReviewBottomSheetRef.current?.snapToIndex(0)
+          break
+        case 'possible':
+          // TODO: show review screen with the possible transactions
```

Comment:
The review screen will use the fully determined TXs to show the maximum gas cost expected. 

And then once the user approves, they would simply be signed and sent to the network, no other logic needed related to fees.